### DISCUSSION TITLE
请问有没有更好的typescript的支持？

### DISCUSSION THREAD

**Discussion by ron-liu** [Serverless-Devs/Serverless-Devs#397] [Q&A] [answered]
现在我看到的使用typescript的例子主要来自`fc-custom-typescript-event`。

从这个例子看，主要是使用`ts-node`来实现对typescript的支持。

请问`ts-node`是否可以用在production中？我现在找到的答案是[这里](https://github.com/TypeStrong/ts-node/issues/104)，大意是可以，但要加上`--transpile-only`的标签。不过我不确定的是，即使这样在servereless场景下使用`ts-node`会不会影响冷启动或者热启动的速度？

**Comment by heimanba** [comment, marked as answer]
建议编译成js 执行，本质是是运nodejs环境上，nodejs原生不支持ts。理论上会影响冷启动速度，因为毕竟你加了一个ts-node的三方包。不过TS-node的性能不太好。
